internal/ini/ini_parser.go

```diff
@@ -242,7 +247,7 @@ loop:
 				}
 
 				children[len(children)-1] = rhs
-				k.SetChildren(children)
+				root.SetChildren(children)
```

Comment:
I am curious about what this change does.

Was this previously setting children on incorrect parent?

Reply:
Nit: Does the comment at https://github.com/aws/aws-sdk-go-v2/pull/1324/files#diff-42e49a0dfb77920f8deb63d9b2902c1c276fa955fb8089128a242adccf7d76feR210-R224 need an update too?

I am fine with just asking to look at doc.go

Reply:
Yeah I'd vote to remove this and not spread the grammar all about.

> Was this previously setting children on incorrect parent?

I took this as a bug (this code wasn't really exercised by the existing test cases before cause they would never hit this block). I was seeing that the children were also being added to the AST identified by K which didn't seem to make sense to me. The intention of this block seemed to be that you get the RHS from the children list of `root`, modify the RHS and set it back on `children` and ensure that the reference on root to it's children was still correct. Where as this was updating K with root's children.